plugins/arrow/io/ArrowReader.cpp

```diff
@@ -0,0 +1,532 @@
+        if (dArray)
+        {
+            point.setField<double>(m_arrayIds[columnNum], dArray->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::FloatArray* fArray = dynamic_cast<arrow::FloatArray*>(array.get());
+        if (fArray)
+        {
+            point.setField<float>(m_arrayIds[columnNum], fArray->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::Int8Array* int8Array = dynamic_cast<arrow::Int8Array*>(array.get());
+        if (int8Array)
+        {
+            point.setField<int8_t>(m_arrayIds[columnNum], int8Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::UInt8Array* uint8Array = dynamic_cast<arrow::UInt8Array*>(array.get());
+        if (uint8Array)
+        {
+            point.setField<uint8_t>(m_arrayIds[columnNum], uint8Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::Int16Array* int16Array = dynamic_cast<arrow::Int16Array*>(array.get());
+        if (int16Array)
+        {
+            point.setField<int16_t>(m_arrayIds[columnNum], int16Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::UInt16Array* uint16Array = dynamic_cast<arrow::UInt16Array*>(array.get());
+        if (uint16Array)
+        {
+            point.setField<uint16_t>(m_arrayIds[columnNum], uint16Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::Int32Array* int32Array = dynamic_cast<arrow::Int32Array*>(array.get());
+        if (int32Array)
+        {
+            point.setField<int32_t>(m_arrayIds[columnNum], int32Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::UInt32Array* uint32Array = dynamic_cast<arrow::UInt32Array*>(array.get());
+        if (uint32Array)
+        {
+            point.setField<uint32_t>(m_arrayIds[columnNum], uint32Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::Int64Array* int64Array = dynamic_cast<arrow::Int64Array*>(array.get());
+        if (int64Array)
+        {
+            point.setField<int64_t>(m_arrayIds[columnNum], int64Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::UInt64Array* uint64Array = dynamic_cast<arrow::UInt64Array*>(array.get());
+        if (uint64Array)
+        {
+            point.setField<uint64_t>(m_arrayIds[columnNum], uint64Array->Value(m_currentBatchPointIndex));
+            continue;
+        }
+        arrow::BinaryArray* binaryArray = dynamic_cast<arrow::BinaryArray*>(array.get());
+        if (binaryArray)
+        {
+            // Binary arrays are assumed to be WKT from GeoParquet
+            std::string_view wkb = binaryArray->Value(m_currentBatchPointIndex);
+            pdal::Geometry pt = pdal::Geometry(std::string(wkb));
+            OGRGeometry* g = (OGRGeometry*) pt.getOGRHandle();
+            OGRPoint* p = g->toPoint();
```

Comment:
toPoint() is actually a static_cast, and thus you should only invoke it if you known that wkbFlatten(g->getGeometryType()) == wkbPoint. Or use a dynamic_cast<OGRPoint*>(g)